rancher2/structure_node_template.go

```diff
@@ -2,10 +2,27 @@ package rancher2
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 )
 
+func fixNodeTempateID(current, id string) string {
+	if len(current) == 0 || len(id) == 0 {
+		return ""
+	}
+	// Fix breaking API change https://github.com/rancher/rancher/pull/23718
+	limitVersion := "2.3.3"
+	if ok, _ := IsVersionGreaterThanOrEqual(current, limitVersion); ok {
```

Comment:
you added IsRancherVersionLessThan and IsRancherVersionGreaterThanOrEqual and then didn't use them and used your util function IsVersionGreaterThanOrEqual which you use in the two rancher version checkers too.. do one or the other maybe? you woulndt need to pass in `current` and could do everything inside this function. if you dont want to do that then maybe remove the two unused functions unless you have a purpose for them in the future.

Reply:
You are right. Updated PR.